src/Plugin/QuickForm/SaatenUnionSpraying.php

```diff
@@ -0,0 +1,746 @@
+      '#required' => TRUE,
+      '#multiple' => TRUE,
+    ];
+
+    $equipment_options = $this->getEquipmentOptions();
+    $form['equipment'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Equipment'),
+      '#description' => $this->t('Select the tractor, sprayer and nozzle equipment assets used for this spraying'),  
+      '#options' => $equipment_options,
+      '#required' => TRUE,
+      '#multiple' => TRUE,
+    ];
+
+    $form['justification_target'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Justification/Target'),
+      '#description' => $this->t('The reason the operation is necessary, and any target pest(s) where applicable.'),
+      '#required' => TRUE,
+    ];
+
+    $form['plant_growth_stage'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Plant growth stage'),
+      '#description' => $this->t('The plant growth stage when the product was applied.'),
+      '#required' => FALSE,
+    ];
+
+    $form['equipment_rinsed'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Equipment triple rinsed'),
+      '#description' => $this->t('Select if the equipment was triple rinsed after the job was completed.'),
+      '#return_value' => 'Yes',
+    ];
+
+    $form['equipment_clear'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Equipment All Clear'),
+      '#description' => $this->t('Select if the equipment all clear after the job was completed.'),
+      '#return_value' => 'Yes',
+    ];
+
+    $form['equipment_washed'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Equipment Clear Washed'),
+      '#description' => $this->t('Select if the equipment was clear washed after the job was completed.'),
+      '#return_value' => 'Yes',
+    ];
+
+    // Log notes.
+    $form['notes'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Notes'),
+      '#description' => $this->t('Any additional notes.'),
+      '#weight' => 20,
+    ];
+    
+    return $form;
+  }
+
+  /**
+  * Submit handler for Saaten Union Spraying Quick Form.
+  */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+
+  // Quantities.
+  $quantity_keys = [
+    'product_rate', 
+    'area',
+    'temperature',
+    'wind_speed',
+    'water_volume',
+    'total_product_quantity',
+    'pressure',
+    'speed_driven'
+  ];
+  $quantities = $this->getQuantities($quantity_keys, $form_state);
+
+  // Notes.
+  $note_fields[] = [
+    'key' => 'notes',
+    'label' => $this->t('Additional notes'),
+  ];
+  $notes = $this->prepareNotes($note_fields, $form_state);
+
+  // Input log entity.
+  $this->createLog([
+      'type' => "input",
+      'name' => $form_state->getValue('log_name'),
+      'timestamp' => $form_state->getValue('start_date')->getTimestamp(),
+      'status' => $form_state->getValue('status'),
+      'flag' => $form_state->getValue('flag'),
+      'assigned_to' => $form_state->getValue('assigned_to'),
```

Comment:
Ah, this is just a confusing one...  logs don't have an `assigned_to` field, it is simply called `owner`.